llvm/lib/TargetParser/ARMTargetParser.cpp

```diff
@@ -600,11 +601,17 @@ StringRef ARM::getARMCPUForArch(const llvm::Triple &Triple, StringRef MArch) {
   llvm_unreachable("invalid arch name");
 }
 
-void ARM::PrintSupportedExtensions() {
+void ARM::PrintSupportedExtensions(std::map<StringRef, StringRef> llvmDescMap) {
   outs() << "All available -march extensions for ARM\n\n";
   for (const auto &Ext : ARCHExtNames) {
     // Extensions without a feature cannot be used with -march.
-    if (!Ext.Feature.empty())
-      outs() << '\t' << Ext.Name << "\n";
+    if (!Ext.Feature.empty()) {
+      if (llvmDescMap.empty()) {
+        outs() << '\t' << Ext.Name << "\n";
+      } else {
+        outs() << format("\t%-20s", Ext.Name.str().c_str());
+        outs() << "\t\t" << llvmDescMap[Ext.Name] << "\n";
```

Comment:
This can be one call to outs().

Reply:
> This can be one call to outs().

The ```if-else``` It is due to separation of "Description" column as optional.
To compress the 2*outs (inside else) format then will be quite long expansion: 
```llvmDescMap["experimental-"+Ext.Name].str().c_str() ```

Will try again, but spent quite some time on this very estetic part.

Reply:
Sorry, I think I wasn't clear, I meant this could be one call:
```
outs() << format("\t%-20s", Ext.Name.str().c_str());
outs() << "\t\t" << llvmDescMap[Ext.Name] << "\n";
```
As in:
```
outs() << format("\t%-20s", Ext.Name.str().c_str())
          << "\t\t" << llvmDescMap[Ext.Name] << "\n";
```

I didn't mean that having an if/else was a bad idea.